client/state/stats/lists/selectors.js

```diff
@@ -201,9 +201,20 @@ export function getSiteStatsViewSummary( state, siteId ) {
 		viewSummary[ years ][ months ].data.push( item );
 		const average =
 			viewSummary[ years ][ months ].total / viewSummary[ years ][ months ].daysInMonth;
-		viewSummary[ years ][ months ].average = round( average, 0 );
+		viewSummary[ years ][ months ].average = Math.round( average );
 	} );
 
+	// With the current month, calculate the average based on the days passed so far in the month
+	if ( viewData.data.length ) {
+		const date = new Date();
+		const thisMonth = viewSummary[ date.getFullYear() ][ date.getMonth() ];
```

Comment:
It's possible that there won't be any stats for this year. In that case, `viewSummary[ date.getFullYear() ]` will be undefined, causing `viewSummary[ date.getFullYear() ][ date.getMonth() ]` to fail with an error like this:

![](https://cldup.com/kq8BapbRLQ.png)

We should try to access this safely, and if there are no stats at all, we don't have to do any calculations.

Reply:
Thank you! I've added a commit that should fix that issue.